Removing the check about the size re: ray-project/ray#3450

devin-petersohn opened the pull request:
<!--
Thank you for your contribution!

Please review https://github.com/ray-project/ray/blob/master/CONTRIBUTING.rst before opening a pull request.
-->
I have opened this in the interest of continuing the discussion here.

## What do these changes do?

Removes the check causing #3450 

This is important in Modin specifically to allow us to use the disk for plasma.
<!-- Please give a short brief about these changes. -->

## Related issue number
Resolves #3450 
<!-- Are there any issues opened that will be resolved by merging this change? -->

robertnishihara commented:
Please don't merge this. If this happens it is almost always an error.

If we want to allow this then we need to make it's only happening in the case where the user passed in the object store memory and we need to provide a very scary warning.

devin-petersohn commented:
I guess we should decide whether users should be responsible for the results of specifying `object_store_memory`. Was this causing some bugs previously?

I do not think raising a scary warning would be helpful to Modin. Printing the same information that was previously printed would be good enough for our purposes.

pcmoritz commented:
I think the right thing to do here is to compare the object_store_memory with the size of the plasma_directory (for /tmp it would be the size of the disk, for /dev/shm it would be the size of the shared memory pool)

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/9744/
Test FAILed.

devin-petersohn commented:
Updated PR given previous comments. 

It is not possible to verify the size of the device that is mounted for the `plasma_directory` in the general case. 

I moved the check inside the condition `if plasma_directory is None`.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/9780/
Test FAILed.

robertnishihara commented on python/ray/services.py:
```diff
@@ -1028,9 +1028,6 @@ def determine_plasma_store_config(object_store_memory=None,
             object_store_memory = MAX_DEFAULT_MEM
 
-    if plasma_directory is not None:
-        plasma_directory = os.path.abspath(plasma_directory)
```
Why remove this?

robertnishihara replied:
Ok, I see you just moved it.

pcmoritz commented:
jenkins retest this please

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Ray-PRB/9827/
Test FAILed.
